cmd/cmd_test.go

```diff
@@ -662,3 +664,24 @@ func TestBadCaseIsJustWarning(t *testing.T) {
 		t.Errorf("expect no error (just a warning), got %q", err)
 	}
 }
+
+func TestSetupHappensBeforeOtherCommands(t *testing.T) {
+	ctx := context.Background()
+
+	qriHome := createTmpQriHome(t)
+	stdinText := "qri_test_name"
+	cmd, shutdown := newCommandWithStdin(ctx, qriHome, stdinText, repotest.NewTestCrypto())
+
+	cmdTextList := [8]string{"qri add", "qri connect", "qri diff", "qri get", "qri init", "qri list", "qri pull", "qri search"}
```

Comment:
Def welcome some feedback here if there's a better way of handling this -- I ended up listing the commands that users would likely run before realizing they need to set up their profile

Reply:
This works well enough! I don't think we need to exhaustively list everything, especially since the command list will change, and the important thing is to test that the error happens at all, to catch any future change that might break that behavior (more migrations works?).

Two things though:
1) No need to pass anything to "standard in", none of these commands read it. Use `newCommand` instead of `newCommandWithStdin` and get rid of `stdinText`.
2) `qri add` is an alias of `qri pull`, but we're favoring `pull` and want to eventually get rid of `add`. Drop `add` and keep `pull`.

Reply:
Ah gotcha, ok great will change now